service/history/decision/handler_test.go

```diff
@@ -547,6 +653,84 @@ func (s *DecisionHandlerSuite) TestHandleBufferedQueries_QueryTooLarge() {
+				queryRegistry.EXPECT().HasBufferedQuery().Return(false)
+			},
+			assertCalls: func(logs *observer.ObservedLogs) {},
+		},
+		{
+			name: "set query termination state failed - client unsupported",
+			expectMockCalls: func() {
+				queryRegistry := query.NewMockRegistry(s.controller)
+				s.mockMutableState.EXPECT().GetQueryRegistry().Return(queryRegistry)
+				queryRegistry.EXPECT().HasBufferedQuery().Return(true)
+				queryRegistry.EXPECT().GetBufferedIDs().Return([]string{"some-buffered-id"})
+				queryRegistry.EXPECT().SetTerminationState("some-buffered-id", gomock.Any()).Return(&types.InternalServiceError{Message: "query does not exist"})
+			},
+			assertCalls: func(logs *observer.ObservedLogs) {
+				s.Equal(1, logs.FilterMessage("failed to set query termination state to failed").Len())
+			},
+			clientFeatureVersion: "0.0.0",
+		},
+		{
+			name: "set query termination state failed - query too large",
+			expectMockCalls: func() {
+				queryRegistry := query.NewMockRegistry(s.controller)
+				s.mockMutableState.EXPECT().GetQueryRegistry().Return(queryRegistry)
+				queryRegistry.EXPECT().HasBufferedQuery().Return(true)
+				queryRegistry.EXPECT().GetBufferedIDs().Return([]string{"some-id"})
+				queryRegistry.EXPECT().SetTerminationState("some-id", gomock.Any()).Return(&types.InternalServiceError{Message: "query already in terminal state"}).Times(2)
+			},
+			queryResults:         s.constructQueryResults([]string{"some-id"}, 10*1024*1024),
+			clientFeatureVersion: client.GoWorkerConsistentQueryVersion,
+			assertCalls: func(logs *observer.ObservedLogs) {
+				s.Equal(1, logs.FilterMessage("failed to set query termination state to failed").Len())
+				s.Equal(1, logs.FilterMessage("failed to set query termination state to unblocked").Len())
+			},
+		},
+		{
+			name: "set query termination state unblocked",
+			expectMockCalls: func() {
+				queryRegistry := query.NewMockRegistry(s.controller)
+				s.mockMutableState.EXPECT().GetQueryRegistry().Return(queryRegistry)
+				queryRegistry.EXPECT().HasBufferedQuery().Return(true)
+				queryRegistry.EXPECT().GetBufferedIDs().Return([]string{"some-buffered-id"})
+				queryRegistry.EXPECT().SetTerminationState("some-id", gomock.Any()).Return(&types.InternalServiceError{Message: "query does not exist"})
+				queryRegistry.EXPECT().SetTerminationState("some-buffered-id", gomock.Any()).Return(&types.InternalServiceError{Message: "query already in terminal state"})
+			},
+			clientFeatureVersion: client.GoWorkerConsistentQueryVersion,
+			queryResults:         map[string]*types.WorkflowQueryResult{"some-id": &types.WorkflowQueryResult{}},
+			assertCalls: func(logs *observer.ObservedLogs) {
+				s.Equal(1, logs.FilterMessage("failed to set query termination state to completed").Len())
+				s.Equal(1, logs.FilterMessage("failed to set query termination state to unblocked").Len())
+			},
+		},
+	}
+	for _, test := range tests {
+		s.Run(test.name, func() {
```

Comment:
you're kinda using a table test and a suite test at the same time which - normally I'd say is probably a bit confusing, but I take it that there's existing tests in here that you're re-using. One thing that *might* cause problems is that the `s.` may maintain state between tests, (like on line 725) so I'd suggest if possible moving to just use plain `t.Run()` and creating a new decisionHandler every time in future.

Reply:
changed in #5932